src/services/document-service.js

```diff
@@ -96,35 +96,39 @@ class DocumentService {
     return this.documentStore.find(filter, { sort: lastUpdatedFirst, projection: metadataProjection });
   }
 
-  async getDocumentsByTags(searchQuery, { includeArchived } = {}) {
-    const searchTags = new Set(searchQuery.trim()
-      .split(/\s+/)
-      .map(tag => escapeStringRegexp(tag.toLowerCase()))
-      .filter(tag => tag.length > 2));
+  async getDocumentsByTags(searchQuery) {
+    const tokens = searchQuery.trim().split(/\s+/);
 
-    if (!searchTags.size) {
+    const positiveTokens = new Set(tokens
+      .filter(token => !token.startsWith('-'))
+      .filter(token => token.length > 2)
```

Comment:
We could put the `.filter(token => token.length > 2)` up in `tokens` since it filters out invalid tags for all further cases.

Reply:
Ah never mind, does not cover the `-token` cases 🤦

Reply:
But the minus tags have to get the minus removed first before measuring the length?